### PR TITLE
Improve script interaction with pacman

### LINKED ISSUE #16362: scripts/setup-termux.sh expects apt
The script `scripts/setup-termux.sh` expects the package manager to be apt. The script should use `pacman` instead if nedded.

### PR DESCRIPTION
**Pull request by Maxython**
Close #16362

### CODE REVIEW

File: scripts/setup-termux.sh [hunk 1]
```diff
@@ -45,6 +48,13 @@ PACKAGES+=" valac"
...
 
-apt update
-apt dist-upgrade -y
-apt install -y $PACKAGES
+if [ "$TERMUX_APP_PACKAGE_MANAGER" = "apt" ]; then
+	apt update
+	apt dist-upgrade -y
+	apt install -y $PACKAGES
+elif [ "$TERMUX_APP_PACKAGE_MANAGER" = "pacman" ]; then
+	pacman -Sy $PACKAGES --needed --noconfirm
```
Comment (2096779623) [inline scripts/setup-termux.sh:56]: Does it really not need `pacman -Syu`?

File: scripts/setup-termux.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# Setup TERMUX_APP_PACKAGE_MANAGER
+source "/data/data/com.termux/files/usr/bin/termux-setup-package-manager" || true
```
[the remark this answers is not in the document]
Comment (2096779623) [reply scripts/setup-termux.sh:4]: > It should read `scripts/properties.sh` like everything else in this repo and use `$TERMUX_PREFIX`: https://github.com/termux/termux-packages/blob/master/scripts/properties.sh#L38
> 
> `$PREFIX` environment variable shouldn't be used.

https://github.com/termux/termux-packages/wiki/Build-environment#termux

This script does not work in termux if `$TERMUX_PREFIX` is used.
```bash
~/termux-packages/scripts $ ./setup-termux.sh
./setup-termux.sh: line 3: /bin/termux-setup-package-manager: No such file or directory
```

File: scripts/setup-termux.sh (hunk 1 above)
Comment (Maxython) [reply scripts/setup-termux.sh:56]: Yes, I think it would be better to have `pacman -Syu`.

**Comment by Maxython** [comment]
Question: how about replacing `bison` with `byacc` (https://github.com/termux/termux-packages/blob/master/scripts/setup-termux.sh#L21)?

**Pull request merged**